### ISSUE TITLE
liquibase.util.grammar.TokenMgrError when parsing blob/binary Data in 4.22.0

### ISSUE THREAD

**Issue by ojas-krishnaprasad** [liquibase/liquibase#4365]
### Search first

- [X] I searched and no similar issues were found

### Description

When we are inserting blob data through liquibase sql We are getting Token Error. 
[Files.zip](https://github.com/liquibase/liquibase/files/11715395/Files.zip)

### Steps To Reproduce

The Reproducable code was presented [here](https://github.com/ojas-krishnaprasad/Liquibase). You can take the below files and accordingly change the paths and Db connections info. Run the Main.java
[Files.zip](https://github.com/liquibase/liquibase/files/11715395/Files.zip)
We are not experiencing this issue when we are using Liquibase core version 4.9.1.

### Expected/Desired Behavior

The Script has to be executed successfully irrespective of liquidise core version.

### Liquibase Version

4.11.0 -> 4.22.0

### Database Vendor & Version

mysql 8.0.29

### Liquibase Integration

JAVA

### OS and/or Infrastructure Type/Provider

UBUNTU

### Additional Context

IN few days back we received errors w.r.t Liquibase outage, So we try to bumping liquibase from 4.9.1 to 4.22.0 (Latest) As a result we experienced this issue.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by nvoxland** [comment]
Thanks for the test case, @ojas-krishnaprasad. Fun to know mysql lets you load binary data as a quoted string like that.

**Comment by jlarge11** [comment]
Is there a workaround for this, or is it better for us to just hold off on upgrading until a fix is in place?

**Comment by tati-qalified** [comment]
Hi @ojas-krishnaprasad , how are you?
Liquibase version 4.23.0 just came out, and I was able to run the code with no issues using this latest version. I'll be closing this ticket, though be sure to reopen it and contact us if the issue persists.

**Comment by jlarge11** [comment]
Initial testing on my end looks good. Thanks for the quick fix!

**Comment by tati-qalified** [comment]
Hm, that's strange. Would you be able to provide some error codes or logs?
I'll look into it again, and I'll let you know.

**Comment by ojas-krishnaprasad** [comment]
HI @tati-qalified thanks, I am Fine. I am not sure how it was running for you. I tried with the latest version (4.23.0) and still facing the same issue. Error file was attached for your reference
[error.txt](https://github.com/liquibase/liquibase/files/11897931/error.txt)

**Comment by jlarge11** [comment]
One thing I noticed when I played around with this a couple weeks back was that some of the binary text in the dump file we have shows up as single quotes in my text editor, and it seemed like that was tripping things up. Just for experimenting, tried removing them, and then I tried escaping them, and then the problem would no longer shop up.

**Comment by tati-qalified** [comment]
@jlarge11 good catch, I looked at @ojas-krishnaprasad 's binary file and error text, and the thing that is breaking the code seems to be a single quote. I'll get in touch with the development team about this. In the meantime, when possible, escaping single quotes would be a workaround.

**Comment by MalloD12** [comment]
Hi guys, is any of you keen to submit a fix for this? I have used the test project shared here and got exactly the same output you guys have been facing. Have you only escaped the single quotes from the beginning-end of the binary content? I tried that and didn't work for me. 

```
'GIF89a\0\0�\0\0������k��J�������X�����������䊏������廽���������(+luw���LUVx�ϛ��h��?������������v�����!�\0\0\0\0\0,\0\0\0\0\0\0\0��\'�D�U@c�M�,@0e�o?NG��yD����H�<(U���t��ȁ�\`p2��@<�!�(�NB�(P�NZu	\nHuz\nNtHI�^		o\n�,N�	f��^s����\"�����^\0\0	����������q�\r\ZԲ\r\ZHka��F>8� A��LpC\"\0�[h`XAƌ-\Z0�\0�I\0\0�|�\0\0;'
```

**Comment by jlarge11** [comment]
Hi, just checking in. Any updates on this issue?

**Comment by kevin-atx** [comment]
@MalloD12 asked to add this as a Candidate for the next sprint.

**Comment by ojas-krishnaprasad** [comment]
ANy updates on this issue

**Comment by ojas-krishnaprasad** [comment]
Hi, just checking in. Any updates on this issue?

**Comment by mariochampion** [comment]
hey @ojas-krishnaprasad, after looking at this in-depth, there is quite a bit of risk, a need for extensive testing, and not enough urgency for us to take this on. We would be happy to help guide you (or others) through a PR to address this, if you so desire. In the meantime I will close this, but feel free to comment back if you want to discuss further. thanks
